python-package/xgboost/spark/data.py

```diff
@@ -101,11 +102,55 @@ def reset(self) -> None:
         self._iter = 0
 
 
+def _read_csr_matrix_from_unwrapped_spark_vec(part: pd.DataFrame) -> csr_matrix:
+    # variables for constructing csr_matrix
+    csr_indices_list, csr_indptr_list, csr_values_list = [], [0], []
+
+    n_features = 0
+
+    for vec_type, vec_size_, vec_indices, vec_values in zip(
+        part.featureVectorType,
+        part.featureVectorSize,
+        part.featureVectorIndices,
+        part.featureVectorValues,
+    ):
+        if vec_type == 0:
```

Comment:
Please correct me if I'm wrong. now that the missing is 0, do we still really need the sparse vector? per my understanding, if one instance has a missing value, then the whole instance will be removed.

Reply:
@trivialfis Is it true ? If so then training on sparse data makes no sense. Almost all instances will be removed ?

Reply:
I think no, @wbo4958 You can check my test case `test_regressor_with_sparse_optim` and `test_classifier_with_sparse_optim`, every training instance contains missing value "0", but the generated model generates good prediction results.